src/descriptor/lang_id.rs

```diff
@@ -0,0 +1,940 @@
+use core::convert::TryFrom;
+
+impl From<&LangID> for u16 {
+    fn from(lang_id: &LangID) -> Self {
+        *lang_id as u16
+    }
+}
+
+impl From<LangID> for u16 {
+    fn from(lang_id: LangID) -> Self {
+        lang_id as u16
+    }
+}
+
+impl TryFrom<u16> for LangID {
```

Comment:
This duplicates all of the LangIDs and makes it easy to accidentally break things. Why not use a readily-available tool like [`num_enum::TryFromPritive`](https://docs.rs/num_enum/latest/num_enum/derive.TryFromPrimitive.html)?

Reply:
done